gensim/corpora/wikicorpus.py

```diff
@@ -544,10 +551,15 @@ def __init__(self, fname, processes=None, lemmatize=utils.has_pattern(), diction
             Maximal token length.
         lower : bool, optional
              If True - convert all text to lower case.
+        filter_articles: callable, optional
+            If set each XML article element will be passed to this callable before being processed. Only articles
+            where the callable returns an XML element are processed, returning None allows filtering out
+            some articles based on customised rules.
```

Comment:
Are there any restrictions on the callable? Can it be a lambda function? (troublesome with parallelized multiprocessing)

Reply:
True, for the parallel `self.process` to work `filter_articles` needs to be serializable, what's the practice for checking this in `gensim`? Try to `pickle` the callable and raise an informative exception if that fails? Is `pickle` the standard serialiser in `gensim`?

On a related "fun" note, see https://stackoverflow.com/questions/50328386/python-typing-pickle-and-serialisation

Reply:
I was thinking a simple note in the docs. What does the error look like now, if a user passes something not pickleable? If it looks reasonable, I'd just keep that.

@menshikh-iv what is our policy here in general?

Reply:
Well there's no specific catch for this so it'll just be python's `PicklingError` - that plus a note in the docs I think should be enough, but I don't know how this is handled elsewhere in `gensim`

Reply:
I think so too. Someone advanced enough to be setting this parameter better be advanced enough to RTFM :)